source-hubspot/source_hubspot/streams.py

```diff
@@ -461,6 +464,11 @@ def read_records(
                     )
                     records = self._transform(self.parse_response(response, stream_state=stream_state, stream_slice=stream_slice))
 
+                if old_expiry_date != self.authenticator._token_expiry_date:
```

Comment:
I don't understand how this works. A few lines above we are doing a deep-copy of `old_expiry_date` and `self.authenticator._token_expiry_date`.

Either the deep copy is equal to the original value, in which case we will always have `old_expiry_date == self.authenticator._token_expiry_date` or it is not equal, and we will always have the opposite, since I don't see either of these values changing independently from each other. They are always a deep copy of each other. I might be missing some nuance about how the authenticator works, if so, can you please add a comment explaining how this works? Thank you

Reply:
The `self.authenticator` handles a class (`Oauth2Authenticator`) that manages token rotation and the `._token_expiry_date` property. When a token rotates, this property is modified and thus will hold a new value, thats why i deep copy the value and compare it with a probable new one. New comments were added in e10bdb7f5a7456886d7d65d7709a3b1c21a2ff9f hopefully making it better to understand, would really appreciate with your feedback if these new comments help to understand better how the changes work.
Also, i've futher tested this change to 15 hours tests, and updated that on the PR description along with the running stats

Reply:
The `self.authenticator` handles a class (`Oauth2Authenticator`) that manages token rotation and the `._token_expiry_date` property. When a token rotates, this property is modified and thus will hold a new value, thats why i deep copy the value and compare it with a probable new one. New comments were added in e10bdb7f5a7456886d7d65d7709a3b1c21a2ff9f hopefully making it better to understand, would really appreciate with your feedback if these new comments help to understand better how the changes work.
Also, i've futher tested this change to 15 hours tests, and updated that on the PR description along with the running stats

Reply:
Thanks, that makes sense now, the comments are really helpful!